backend/pkg/interactors/update_state.go

```diff
@@ -0,0 +1,35 @@
+package interactors
+
+import (
+	"fmt"
+
+	"github.com/CESARBR/knot-gateway-webui/backend/pkg/data"
+	"github.com/CESARBR/knot-gateway-webui/backend/pkg/logging"
+)
+
+// UpdateStateInteractor represents the update state use case
+type UpdateStateInteractor struct {
+	stateRepository data.Repository
+	logger          logging.Logger
+}
+
+// NewUpdateStateInteractor creates a new UpdateStateInteractor intance
+func NewUpdateStateInteractor(stateRepository data.Repository, logger logging.Logger) *UpdateStateInteractor {
+	return &UpdateStateInteractor{stateRepository: stateRepository, logger: logger}
+}
+
+// Execute runs the use case logic
+func (i *UpdateStateInteractor) Execute(t string) error {
+	state := i.stateRepository.GetCurrent()
+	if err := state.Update(t); err != nil {
```

Comment:
Is that common to make an assignment in an `if` statement?
Isn't it better declare err and later make the assignment?
Like:
```
var err int

err = state.Update(t)
if err != nil { ... }

err = i.stateRepository.Update(state)
```

Reply:
Actually, isn't it better call just `state.Update` inside `i.stateRepository.Update(state)`?

The err comes as a string to be logged

Reply:
I have removed the assignment in the same line of the `if` statement, I think you are right and it becomes clearer. However, about your second suggestion, I think we can't do that because `state.Update()` is a domain operation that is tied with the use case needs instead of the repository ones. Notice the `stateRepository` should remain the same even if it's used by other use cases that don't want to manipulate the entity by calling the `state.Update()` method.

Reply:
BTW, thanks for you suggestions.